File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void get_target(vector<int> numbers, int target, int sum, int index) {    
    if(index == numbers.size()) {
        if(sum == target) {
            answer++;
        }
        return;
    }
    
    get_target(numbers, target, sum + numbers[index], index +1);
    get_target(numbers, target, sum - numbers[index], index +1);
}

int solution(vector<int> numbers, int target) {
    get_target(numbers, target, 0, 0);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cs
using System;

public class Solution {
    int answer = 0;
    public int solution(int[] numbers, int target) {
        GetTarget(numbers, target, 0, 0);
        return answer;
    }
    
    void GetTarget(int[] numbers, int target, int sum, int index) {
        if(index == numbers.Length) {
            if(sum == target) answer++;
            return;
        }
        GetTarget(numbers, target, sum + numbers[index], index + 1);
        GetTarget(numbers, target, sum - numbers[index], index + 1);
        
    }
}